### PR TITLE
refactor: move configuration to pinia store

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
Moves the configuration from a vuex store module to pinia. Also removes the `ConfigurationManager` in favour of the new pinia composable `useConfigStore`.

The benefits on this one are huge:

* No more spread logic between the config store and the `ConfigurationManager`. There is only one single source of truth, which is the new pinia config store.
* Proper types for all configs across the board.
* Easy unit testing as with all pinia composables, no more `jest.mock('.../useConfigurationManager', ...)`.

## Related Issue

- refs https://github.com/owncloud/web/issues/10210

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

### CODE REVIEW

File: packages/web-runtime/src/layouts/Application.vue [hunk 1]
```diff
@@ -208,14 +199,6 @@ export default defineComponent({
...
       })
 
-      // Get extensions manually added into config
-      this.configuration.applications.forEach((application) => {
-        list.push({
-          ...application,
-          type: 'link'
-        })
-      })
-
```
Comment (JammingBen) [inline packages/web-runtime/src/layouts/Application.vue:218]: Not 100% sure, but I guess this was old (oC10) code.

File: packages/web-app-files/src/services/folder/loaderSpace.ts [hunk 2]
```diff
@@ -32,15 +32,8 @@ export class FolderLoaderSpace implements FolderLoader {
...
-      store,
-      userStore,
-      spacesStore,
-      router,
-      clientService,
-      configurationManager,
-      capabilityStore
-    } = context
+    const { store, userStore, spacesStore, router, clientService, configStore, capabilityStore } =
+      context
```
Comment (dschmidt) [inline packages/web-app-files/src/services/folder/loaderSpace.ts:36]: The TaskContext looks really awkward by now - we should revisit it at some point

File: packages/web-pkg/src/composables/piniaStores/config/types.ts
Comment (dschmidt) [inline packages/web-pkg/src/composables/piniaStores/config/types.ts:1]: While at it: why not a zod schema? 🤓

File: packages/web-runtime/src/container/bootstrap.ts [hunk 3]
```diff
@@ -613,23 +597,17 @@ export const startSentry = (runtimeConfiguration: RuntimeConfiguration, app: App
       attachProps: true,
       logErrors: true,
       ...moreSentryOptions
-    })
+    } as SentryConfig)
```
Comment (dschmidt) [inline packages/web-runtime/src/container/bootstrap.ts:600]: ... why is this needed?

File: packages/web-runtime/src/container/bootstrap.ts [hunk 4]
```diff
@@ -613,23 +597,17 @@ export const startSentry = (runtimeConfiguration: RuntimeConfiguration, app: App
...
-export const announceCustomScripts = ({
-  runtimeConfiguration
-}: {
-  runtimeConfiguration?: RuntimeConfiguration
-}): void => {
-  const { scripts = [] } = runtimeConfiguration
-
-  scripts.forEach(({ src = '', async = false }) => {
+export const announceCustomScripts = ({ configStore }: { configStore?: ConfigStore }): void => {
+  ;(configStore.scripts || []).forEach(({ src = '', async = false }) => {
```
Comment (dschmidt) [inline packages/web-runtime/src/container/bootstrap.ts:610]: Couldn't/should empty array be the default?

File: packages/web-runtime/src/container/bootstrap.ts [hunk 5]
```diff
@@ -644,16 +622,10 @@ export const announceCustomScripts = ({
...
-export const announceCustomStyles = ({
-  runtimeConfiguration
-}: {
-  runtimeConfiguration?: RuntimeConfiguration
-}): void => {
-  const { styles = [] } = runtimeConfiguration
-
-  styles.forEach(({ href = '' }) => {
+export const announceCustomStyles = ({ configStore }: { configStore?: ConfigStore }): void => {
+  ;(configStore.styles || []).forEach(({ href = '' }) => {
```
Comment (dschmidt) [inline packages/web-runtime/src/container/bootstrap.ts:628]: same here

**Review by dschmidt — APPROVED** [review]
holy moly, this is great! <3

File: packages/web-pkg/src/composables/piniaStores/config/types.ts
Comment (JammingBen) [reply packages/web-pkg/src/composables/piniaStores/config/types.ts:1]: Sure, are you willing to give a quick introduction? 😄

File: packages/web-app-files/src/helpers/contextualHelpers.ts [hunk 6]
```diff
@@ -134,7 +135,7 @@ const filterContextHelper = (
   data: ContextualHelperData,
   options?: ContextualHelperOptions
 ): ContextualHelperData => {
-  if ((options.configurationManager as any).options.contextHelpersReadMore === false) {
+  if ((options.configStore as ConfigStore).options.contextHelpersReadMore === false) {
```
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-app-files/src/helpers/contextualHelpers.ts:138]: `ContextualHelperData` sits in the ODS and has typed `configStore` as unknown - hence the type casting here. Might be worth checking if we should removed the contextual helper stuff from the ODS though...

File: packages/web-runtime/src/container/bootstrap.ts (hunk 3 above)
Comment (JammingBen) [reply packages/web-runtime/src/container/bootstrap.ts:600]: Without I get: `The types of 'transportOptions.fetchOptions.window' are incompatible between these types. Type 'unknown' is not assignable to type 'null'.`, which weird, because the `SentryConfig` is of the same type as the function's argument. But I don't know how or why this happens.

File: packages/web-runtime/src/layouts/Application.vue (hunk 1 above)
Comment (dschmidt) [reply packages/web-runtime/src/layouts/Application.vue:218]: Looks superfishy at least

File: packages/web-pkg/src/composables/piniaStores/config/types.ts
Comment (dschmidt) [reply packages/web-pkg/src/composables/piniaStores/config/types.ts:1]: sure, I'd love to

File: packages/web-runtime/src/container/bootstrap.ts (hunk 3 above)
Comment (dschmidt) [reply packages/web-runtime/src/container/bootstrap.ts:600]: odd, we can look together

File: packages/web-app-files/src/helpers/contextualHelpers.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (JammingBen) [reply packages/web-app-files/src/helpers/contextualHelpers.ts]: Fixed it by moving the type to this file instead.
